### PR TITLE
feat: new setup of the sample-app-web

### PR DESCRIPTION
**Pull request by wswebcreation**
> **NOTE:** The current sample-app needs to be updated to at least the latest version of React. Secondly we also need to update the structure so it will not become static pages, but pages that can interact with each other so we can _easily_ integrate new features in the future

**Goal of this PR is:**
* upgrade the project to the latest version of ReactJS by using create-react-app (with all it's extra benefits)
* re-use a lot of the existing code, but re-structure is _better_, further code optimisation will be done in different PR's
* implement the first Unit tests AND E2E tests that can be triggered from GH-actions
* provide a setup for GH-pages that can host the old version (just under https://www.saucedemo.com) and the new structured site (under https://www.saudedemo.com/new) so they can be used together which will give colleagues the time to change their tests if there are any breaking changes.

**TODO:**
- [x] Figure out how to have 2 websites running, with a subfolder for the new one, on GitHub Pages
- [x] The new website is not static anymore, see https://create-react-app.dev/docs/deployment/#github-pages for how to run it with GH-pages
- [x] update the actions to build both projects
- [x] come up with a release and migration strategy/date
- [x] implement basic setup for Unit tests and E2E tests on Sauce Labs based on PR's 

**New feature in this PR:**
- If you're not logged in and want to go to a specific page, you will be redirected to the login with a message like for example the below one. This also resulted in an inconsistency in the tests with different browsers where the `sessionStorage` was gone after a page reload (`localStorage` fixed it, but also results in a breaking change, see below. We can also choose to use `cookies`)
image

**Migration suggestion:**
All new code has been added to the `new`-folder so both sites can live next to each other. The GitHub workflow has been adjusted and now deploys the current code to https://www.saucedemo.com, but the new refactored code to https://www.saucedemo.com/new. Both sites can now live next to each other and all code adjustments can be made in the `new`-folder.
This gives us the opportunity for people to already test their code against the https://www.saucedemo.com/new site, giving them the time till for example end of January to make the changes, and then flip the switch (move new to the main url and _old_ to an _old_ folder)

**BREAKING CHANGES:**
- Storage for storing the user data and cart content has been changed from `sessionStorage` to `localStorage` (We can also choose to use `cookies`). When running all tests on all browsers it seemed that all browsers other than Chrome had issues with keeping the `sessionStorage` in the browser after a new page load. This was inconsistently. After setting it to `localStorage` all tests on all browsers passed
- ~Drop support for IE11? It's pretty hard to get the latest version of ReactJS working on IE11. IE11 will also be deprecated in 2021, so should we still support it in our demo app?~ App now supports IE11

**Tests:**
- Running on Sauce Labs with a local build: https://app.eu-central-1.saucelabs.com/builds/55dab56f156f43bd8a9d805b11f8073f
- Running on Sauce Labs with https://www.saucedemo.com/new: https://app.eu-central-1.saucelabs.com/builds/2a75b4d9c7e4472f995a3dfbeb4a61d6

**Roadmap:**
- [ ] implement unit tests and refactor the code in such a way that we have smaller components without breaking the current workflow, this would also include some UI tests, see https://github.com/saucelabs/sample-app-web/issues/24
- [ ] implement the proper design of the app, without implementing too many breaking changes
- [ ] implement storybook for all components
- [ ] add lazy loading of content, see https://github.com/saucelabs/sample-app-web/issues/32
- [ ] many more

### CODE REVIEW

**Review by L1fescape-sl — APPROVED** [review]
Overall LGTM! just a few minor nit-picks.

File: new/src/pages/Login.css [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  cursor: pointer;
+}
+.login-button:active {
+  background: #ccc;
+}
+.form_input {
+  margin: 0 0 25px 0;
+  font-size: 18px;
+  width: 100%;
+  max-width: 380px;
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply new/src/pages/Login.css:57]: Thanks!

File: new/src/components/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import SwagBotFooter from "../assets/img/SwagBot_Footer_graphic.png";
+import "./Footer.css";
+
+function SwagLabsFooter() {
+  return (
+    <footer className="footer">
+      <ul className="social">
+        <li className="social_twitter">Twitter</li>
+        <li className="social_facebook">Facebook</li>
+        <li className="social_linkedin">LinkedIn</li>
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply new/src/components/Footer.jsx:11]: Thanks, the original didn't had it, but added it now

File: new/src/components/CartItem.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+              className="btn_secondary cart_button"
+              onClick={() => removeFromCart(item.id)}
+            >
+              REMOVE
+            </button>
+          </div>
+        </div>
+      </div>
+    );
+  } else {
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply new/src/components/CartItem.jsx:58]: Fixed it, thanks

File: new/src/components/InventoryListItem.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    if (isProblemUser()) {
+      // Bail out now, don't remove from cart if the item ID is even
+      if (itemId % 2 === 0) {
+        return;
+      }
+    }
+
+    ShoppingCart.removeItem(itemId);
+    setItemInCart(false);
+    console.log(ShoppingCart.getCartContents());
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply new/src/components/InventoryListItem.jsx:46]: Thanks, they were already there, but removed them now

**Pull request merged**